src/templates/gderived-defs

```diff
@@ -156,13 +156,19 @@ define: (ClassBodyDeclarations)ctr_userdict
 define: (ClassBodyDeclarations)toString
   public String toString() {
     PyType self_type = getType();
-    PyObject impl = self_type.lookup("__repr__");
-    if (impl != null) {
-      PyObject res = impl.__get__(this,self_type).__call__();
-      if (!(res instanceof PyString))
+    PyObject impl = self_type.lookup("toString");
+    if (impl != null && impl != TYPE.lookup("toString")) {
+      Object res = impl.__get__(this,self_type).__call__();
```

Comment:
Can `__call__` really produce something other than a `PyObject`? I don't see how. And so some of these added clauses don't apply.